bin/ci/run-tests.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set -eou pipefail
+
+# Don't attempt to build the following plans. They have resource requirements
+# or build times that exceed the currently available resources on our CI
+# infrastructure.
+PLAN_BLACKLIST=(
+ glibc
+ gcc
+ ghc
+ ghc710
+ ghc710-bootstrap
+ ghc80
+ ghc82
+ ghc82-bootstrap
+ ghc84
+ ghc86
+ kubernetes
+ mongodb
+ mysql
+)
+plan="$(basename "$1")"
+
+HAB_ORIGIN=ci
+export HAB_ORIGIN
+
+for bl in "${PLAN_BLACKLIST[@]}"; do
+  if [[ "${plan}" == "${bl}" ]]; then
+    echo "--- :bangbang: [$plan] Skipping build"
+    # If we're running in buildkite, we also want to annotate the build
+    if [[ "${CI:-}" == "true" ]]; then
+      buildkite-agent annotate --style 'warning' ":bangbang: ${plan} found in build blacklist. Skipping build. "
+    fi
+    exit 0;
+  fi
+done
+
+if [ ! -f "$plan/tests/test.sh" ]; then
+  buildkite-agent annotate --style 'warning' ":warning: :linux: ${plan} has no Linux tests to run. This is currently permitted but please consider adding some tests."
+  exit 0
+fi
+
+echo "--- :key: Generating fake origin key"
+# This is intended to be run in the context of public CI where
+# we won't have access to any valid signing keys.
+hab origin key generate "$HAB_ORIGIN"
+
+
+echo "--- :construction: Build $plan"
+
+# We want to ensure that we build from the project root. This
+# creates a subshell so that the cd will only affect that process
+project_root="$(git rev-parse --show-toplevel)"
+(
+  cd "$project_root"
+  hab pkg build "$plan"
+  source results/last_build.env
```

Comment:
I think we should put "--- :mag: Testing $pkg_ident" here to split the logs in buildkite. It will give us a nice fold between build and test.  (https://github.com/buildkite/emojis for finding available emojis if you want to find something more appropriate)

Reply:
:mag: looks good, the only alternative I see there is :slot_machine: which might be a bit too sardonic.

Reply:
Added this logging entry, and also moved the log line for building inside this block, immediately before the build.